### ISSUE TITLE
EmailAddress rule gives false positive

### ISSUE THREAD

**Issue by derskythe** [FluentValidation/FluentValidation#1643]
Hi, found false positive email check.
The code below returns all tests as passed. 
However, in the first inline block (elchın@somedomain.com) has an incorrect character "ı" (Unicode: U+0131)

FluentValidation 9.5.1
.NET 5

```
[Theory]
[InlineData("elchın@somedomain.com", true)]
[InlineData("elchin@somedomain.com", true)]
[InlineData("elchin", false)]
public void CheckEmail(string email, bool isValid)
{
    var validator = new WrongEmailValidator();
    var validationResult = validator.Validate(email);
    Assert.Equal(isValid, validationResult.IsValid);
}

public class WrongEmailValidator : AbstractValidator<string>
{
    public WrongEmailValidator()
    {
        RuleFor(i => i).EmailAddress();
    }
}
```

**Comment by JeremySkinner** [comment]
Hi, this behaviour is correct - the email validator is intentionally naive, and uses the same logic as ASP.NET Core's default EmailAddress attribute for consistency, which is just to check that a string contains an "@" character. Please review these sections in the documentation for details:

https://docs.fluentvalidation.net/en/latest/upgrading-to-9.html#default-email-validation-mode-changed
https://docs.fluentvalidation.net/en/latest/built-in-validators.html#email-validator

You might also find this discussion of interest: https://www.reddit.com/r/csharp/comments/hlzpwn/why_is_emailaddress_attribute_validating_as_true/
